workflow/executor/executor.go

```diff
@@ -267,11 +267,13 @@ func (we *WorkflowExecutor) LoadArtifacts(ctx context.Context) error {
 func (we *WorkflowExecutor) StageFiles() error {
 	var filePath string
 	var body []byte
+	mode := 0o644
```

Comment:
this is failing to build due to a typecheck error (see [lint](https://github.com/argoproj/argo-workflows/actions/runs/8254003171/job/22577149813?pr=12787) failure etc).

I think this is supposed to be written as:
```suggestion
	mode := os.FileMode(0o644)
```

I was hoping there might be some constants for these modes, but seems like [no](https://go.dev/src/io/fs/fs.go) (note that [`os.FileMode = fs.FileMode`](https://pkg.go.dev/os#FileMode))